framework/src/base/FEProblemBase.C

```diff
@@ -364,6 +364,8 @@ void FEProblemBase::initialSetup()
   if (_app.isRecovering() && (_app.isUltimateMaster() || _force_restart))
     _resurrector->setRestartFile(_app.getRecoverFileBase());
 
+  _mesh.initialSetup();
```

Comment:
That of course is the trickiest part of this whole thing.  I put it right before the first place where anything gets called on _mesh.  I'd appreciate feedback on this. For our application, I think we're pretty flexible, but looking at what's going on here, it should probably be called as early as possible.